[Design2-239, Design2-213] Copy TextField component to DSCO package, use semantic colors in TextField component

levadadenys opened the pull request:
<!--
  A summary of the change, including any relevant background, motivation, and context.
  If relevant, include a description, screenshots, and/or video of the existing and new behavior.
-->
### [Design2-239, Design2-213] Copy TextField component to DSCO package, use semantic colors in TextField component

feat(textField): copied textField to dsco/textField
feat(textField): used semantic dsco colors instead of old colors for dsco/textField

## Links
- jira ticket: [Design2-239](https://codedotorg.atlassian.net/browse/DESIGN2-239),  [Design2-213](https://codedotorg.atlassian.net/browse/DESIGN2-213)

<!--
  Links to relevant external resources; ie, specification documents, Jira tickets, related PRs, Honeybadger errors, etc.
-->

<!--
- spec: []()
- jira ticket: []()
-->

## Testing story

<!--
  Does your change include appropriate tests?
  If so, please describe how the tests included in this PR are sufficient.
  If not, please explain why this change does not need to be tested.
-->

<!-- Other aspects to consider. Delete any sections that are not relevant to your change. -->

## Deployment strategy

## Follow-up work

<!--
  List (ideally with Jira links) any clean-up or technical debt that will be addressed in future work.
-->

## Privacy

<!--
  1.	Does this change involve the collection, use, or sharing of new Personal Data?
  2.	Does this change involve a new or changed use or sharing of existing Personal Data?
-->

## Security

<!-- Link to Jira task(s) where sensitive security issues are discussed privately. -->

## Caching

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

levadadenys commented:
By this point, remaining falling storybook ui tests - are caused by TextField error message contrast ration error. Designs are approved by design team so we should be ok to continue using it still. Worth double checking with design team though.

For more context, here's the full error message:
'Element has insufficient color contrast of 4.38 (foreground color: #e5311a, background color: #ffffff, font size: 10.5pt (14px), font weight: normal). Expected contrast ratio of 4.5:1'

kelbyhawn reviewed (commented):
I'm only seeing the following props locally, is this intended?
<img width="1041" alt="Screenshot 2025-01-06 at 12 31 49 PM" src="https://github.com/user-attachments/assets/c1dd6a4f-08c3-4229-9753-46ae3386834e" />

levadadenys commented:
> I'm only seeing the following props locally, is this intended? <img alt="Screenshot 2025-01-06 at 12 31 49 PM" width="1041" src="https://private-user-images.githubusercontent.com/9256643/400545947-c1dd6a4f-08c3-4229-9753-46ae3386834e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyNDI1MzgsIm5iZiI6MTczNjI0MjIzOCwicGF0aCI6Ii85MjU2NjQzLzQwMDU0NTk0Ny1jMWRkNmE0Zi0wOGMzLTQyMjktOTc1My00NmFlMzM4NjgzNGUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMDdUMDkzMDM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDY3YTZjMGVjOTkxZDI4OGFlMjk0NDNkNjA2MjU0NjI1OTJmNmY5NThhNjc5ZTIwYzNjNzhjNWNhNjg3ZDY4YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.1sxNqr3gTYBmIdSVavDG_bzLOYE8_I4TF-fVlUe58nw">

Not intended. Fixed it now. Also spotted that cursor and disabled and readonly state wasn't styles as needed and fixed it now. Also removed Changelog from textField folder since we don't need it
